### ISSUE TITLE
[Idea] Strategies for concurrency 

### ISSUE THREAD

**Issue by MerrionComputing** [MerrionComputing/Presentations#5]
Business and technology solutions to concurrency issues (like the traditional example of the bank transfer).

- Compensation actions
- As-of read model actions
- State change prevention

**Comment by MerrionComputing** [comment]
Why do we need solutions to concurrency issues?

Is "fingers crossed" ever a viable strategy?

**Comment by ylorph** [comment]
Yeah "finger crossed" is viable if it costs less than solving them ;-)

**Comment by skleanthous** [comment]
Or a runbook with "step 1: shout in the room if anyone else is updating global settings. If no-one answers within 2 minutes move to step 2."

**Comment by MerrionComputing** [comment]
You could have a semaphore (the "I am updating these settings" hat...and there is only one hat) - this would be a form of state change prevention.

**Comment by ylorph** [comment]
you mean an Xclusive lock :-)

**Comment by skleanthous** [comment]
Well, yeah it is a kind of semaphore, but I mentioned it because it's outside of the system itself and relies on people playing nice and adhering to a workflow. The effort involved is at least one order of magnitude less than any state change prevention in code.

**Comment by MerrionComputing** [comment]
Working title "What to do if transactions don't exist"...
